cmd/client_test.go

```diff
@@ -0,0 +1,213 @@
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
+		if err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(probe), &myprobe.Probe{}); err != nil {
+			t.Fatalf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+		}
+
+		// Check if the probe was added correctly
+		listProbesResp, err := client.ListProbes(testHermes.Ctx)
+		if err != nil {
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+		}
+
+		probesList := listProbesResp.GetProbe()
+
+		// Sorts the probes by their extension number as they are in the format:
+		// "testExtension" + <number>, e.g. "testExtension0".
+		sort.SliceStable(probesList[:], func(i, j int) bool {
+			probeNum0, _ := strconv.Atoi(probesList[i].GetName()[13:])
+			probeNum1, _ := strconv.Atoi(probesList[j].GetName()[13:])
+			return probeNum0 < probeNum1
+		})
+
+		if probesList[i].GetName() != probe {
+			t.Errorf("Expected probe %s to be running, got probe %s running", probe, probesList[i].GetName())
+		}
+	}
+}
+
+// TestRemoveProbe tests that the RemoveProbe() method removes a probe from Cloudprober
+func TestRemoveProbe(t *testing.T) {
+	testHermes, client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, testHermes, client)
+
+	for i := 0; i < testRunCount; i++ {
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
+		if err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(probe), &myprobe.Probe{}); err != nil {
+			t.Fatalf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+		}
+
+		// Check if the probe was added correctly
+		listProbesResp, err := client.ListProbes(testHermes.Ctx)
+		if err != nil {
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+		}
+
+		probesList := listProbesResp.GetProbe()
+		if probesList[0].GetName() != probe {
+			t.Errorf("Expected probe %s to be running, got probe %s running", probe, probesList[0].GetName())
+		}
+
+		if err = client.RemoveProbe(probe, testHermes.Ctx); err != nil {
+			t.Fatalf("RemoveProbe() failed, expected error: %v, got %v", nil, err)
+		}
+
+		// Check if the probe was removed correctly
+		listProbesResp, err = client.ListProbes(testHermes.Ctx)
+		if err != nil {
+			t.Fatalf("ListProbes() failed, expected error: %v, got: %v", nil, err)
+		}
+
+		probesList = listProbesResp.GetProbe()
+
+		if len(probesList) != 0 {
+			t.Errorf("Expected no probes to be running, got %d probes running; active probes running: %v", len(probesList), probesList)
+		}
+	}
+}
+
+// TestListProbes tests that the ListProbes() method returns all active probes in Cloudprober
+func TestListProbes(t *testing.T) {
+	testHermes, client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, testHermes, client)
+
+	for i := 0; i < testRunCount; i++ {
+		probe := fmt.Sprintf("%s%d", "testExtension", i)
```

Comment:
Should we call this instead `probeName` rather than just `probe`? It feels more descriptive this way.

(This is also applicable in the function above)

Reply:
Yes, that is more descriptive. I'll change it to that.